pykwalify/core.py

```diff
@@ -537,6 +537,9 @@ def _validate_scalar(self, value, rule, path, done=None):
         log.debug(u" # %s", rule.type)
         log.debug(u" # %s", path)
 
+        if rule.allownone and value is None:
+            return True
```

Comment:
I am not 100% sure about having this at the top of the validation. If you look just below it, there is a handler for `func` keyword and even if you have `allownone = True` you might still want funcs to run. There could even be more flags that should be runned even if `allownone = True`. I have to think some about this one and what should be possible/allowed and what shold not.